skglm/experimental/smooth_quantile_regressor.py

```diff
@@ -0,0 +1,553 @@
+    problem. This approach is particularly effective for large datasets where
+    direct optimization of the non-smooth objective can be challenging.
+
+    The solver automatically selects the best solution from all smoothing stages
+    based on the quantile error, ensuring good approximation of the target
+    quantile level.
+
+    References
+    ----------
+    Chen, C. (2007). A Finite Smoothing Algorithm for Quantile Regression.
+    Journal of Computational and Graphical Statistics, 16(1), 136–164.
+    http://www.jstor.org/stable/27594233
+
+    Examples
+    --------
+    >>> from skglm.experimental.smooth_quantile_regressor import SmoothQuantileRegressor
+    >>> import numpy as np
+    >>> X = np.random.randn(1000, 10)
+    >>> y = np.random.randn(1000)
+    >>> reg = SmoothQuantileRegressor(quantile=0.8, alpha=0.1)
+    >>> reg.fit(X, y)
+    >>> print(reg.coef_)
+    """
+
+    def __init__(
+        self,
+        smoothing_sequence=None,
+        quantile=0.5,
+        alpha=0.1,
+        initial_delta=0.5,
+        min_delta=1e-6,
+        smooth_solver=None,
+        verbose=False,
+        delta_tol=1e-6,
+        max_stages=8,
+        quantile_error_threshold=0.005,
+        solver_params=None,
+        initial_alpha: float = None,
+        alpha_schedule: str = 'geometric',
+    ):
+        self.quantile = float(quantile)
+        if not 0 < self.quantile < 1:
+            raise ValueError("quantile must be between 0 and 1")
+        self.alpha = float(alpha)
+        self.initial_delta = float(initial_delta)
+        self.min_delta = float(min_delta)
+        self.verbose = verbose
+        self.delta_tol = float(delta_tol)
+        self.max_stages = int(max_stages)
+        self.quantile_error_threshold = float(quantile_error_threshold)
+        self.smoothing_sequence = smoothing_sequence
+
+        # L1‐penalty continuation: start at initial_alpha, end at self.alpha
+        self.initial_alpha = float(
+            initial_alpha) if initial_alpha is not None else 5.0 * self.alpha
+        self.alpha_schedule = alpha_schedule
+
+        self.solver_params = {} if solver_params is None else dict(solver_params)
+
+        # Initialize solver strategy
+        self.solver_strategy = StageBasedSolverStrategy(self.solver_params)
+
+        from skglm.experimental.quantile_huber import QuantileHuber
+        self._quantile_huber_cls = QuantileHuber
+
+        # Initialize solver - use PDCD for L1 regularization
+        if smooth_solver is None:
+            self.smooth_solver = PDCD_WS(
+                max_iter=200,
+                tol=1e-8,
+                fit_intercept=False,
+                warm_start=True,
+                p0=200
+            )
+        else:
+            self.smooth_solver = smooth_solver
+
+        # Ensure consistent solver settings
+        if not hasattr(self.smooth_solver, 'warm_start'):
+            self.smooth_solver.warm_start = False
+
+    def _initialize_pdcd(self, X, y):
```

Comment:
for a v0, use only smooth problems, so use AndersonCDWS solver instead. It should not have any params